core/environment/environment.go

```diff
@@ -105,6 +107,14 @@ func newEnvironment(userVars map[string]string) (env *Environment, err error) {
 		callsPendingAwait: make(map[string]callable.CallsMap),
 	}
 
+	if envName, ok := userVars["environmentName"]; ok {
```

Comment:
Variables within the environment template context should use `snake_case`, so `environment_name`.